src/Hedgehog/Linq/Property.fs

```diff
@@ -184,4 +139,40 @@ type PropertyExtensions private () =
                     Property.ofThrowing projection.Invoke (a, b)))
         Property result
 
+    [<Extension>]
+    static member WithShrinks (property : Property, shrinkLimit : int<shrinks>) : Property =
+        let (Property property) = property
+        Property (Property.withShrinks shrinkLimit property)
+
+    [<Extension>]
+    static member WithoutShrinks (property : Property) : Property =
+        let (Property property) = property
+        Property (Property.withoutShrinks property)
+    
+    [<Extension>]
+    static member WithTests (property : Property, testLimit : int<tests>) : Property =
+        let (Property property) = property
+        Property (Property.withTests testLimit property)
+    
+    [<Extension>]
+    static member WithConfig (property : Property, config : PropertyConfig) : Property =
+        let (Property property) = property
+        Property (Property.withConfig config property)
```

Comment:
I think we can get rid of this, and only expose configuration with `Property.with*`

Reply:
This is the same as `Property.Config.set` (as I strucutured/named it in another comment).  I think it is better to keep.  Might want to consider renaming to `setConfig` though.

Reply:
What would the value be in overriding a user's preferences?

Reply:
It would be the user calling this function, it would be their preference.  For example, @dharmaturtle talks about how he runs PB tests with a time limit during development and with a test limit during integration.  I think the minimal amount of duplicated code to achieve this would be constructing the correct `PropertyConfig` instance in a global location and then referencing it from all the PB tests in order to set it, which (when using C#) would be achieved via this method.

Reply:
You could also make an extension method of `Property` that configures them to all be the same and call that on each property you want it to apply to.

```csharp
public static Property ConfigureSpecifically(this Property property) =>
    property.WithTests(1000).WithoutShrinkLimit().Whatever();

// in a test

var someProperty = // ...
someProperty.ConfigureSpecifically().OverrideLocally().Check();
```

Reply:
Good point.

I think I would still keep this myself, but I won't argue further in favor of keeping it.